tests/backup/backup_sse_test.go

```diff
@@ -0,0 +1,412 @@
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		providers := getProviders()
+
+		for _, provider := range providers {
+
+			Step("Validate applications", func() {
+				log.InfoD("Validate applications")
+				ValidateApplications(scheduledAppContexts)
+			})
+			Step("Register cluster for backup", func() {
+				log.InfoD(fmt.Sprintf("Creating source and destination cluster"))
+				err = CreateApplicationClusters(orgID, "", "", ctx)
+				dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+				clusterStatus, err = Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+				log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+				dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+				clusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+			})
+			// Create bucket without deny policy
+			Step("Create bucket without deny policy", func() {
+				log.InfoD(fmt.Sprintf("Create bucket without deny policy"))
+				bucketWithOutPolicy := "sse-bucket-without-deny-policy"
+				customBucketWithOutPolicy = GetCustomBucketName(provider, bucketWithOutPolicy)
+				// Update custom bucket for cleanup purpose
+				customBuckets = append(customBuckets, customBucketWithOutPolicy)
+			})
+			// Create bucket with deny policy
+			Step("Create bucket with deny policy", func() {
+				log.InfoD(fmt.Sprintf("Create bucket with deny policy"))
+				numberOfBucketsWithDenyPolicy := 2
+				bucketWithDenyPolicy := "sse-bucket-with-deny-policy"
+				for i := 0; i < numberOfBucketsWithDenyPolicy; i++ {
+					customBucketWithDenyPolicy := GetCustomBucketName(provider, bucketWithDenyPolicy)
+					policy, err := GenerateS3BucketPolicy("DenyNonAES256Uploads", "s3:x-amz-server-side-encryption=AES256", customBucketWithDenyPolicy)
+					log.FailOnError(err, "Failed to generate s3 bucket policy check for the correctness of policy parameters")
+					err = UpdateS3BucketPolicy(customBucketWithDenyPolicy, policy)
+					log.FailOnError(err, "Failed to apply bucket policy")
+					customBucketsWithDenyPolicy = append(customBucketsWithDenyPolicy, customBucketWithDenyPolicy)
+					customBuckets = append(customBuckets, customBucketWithDenyPolicy)
+					log.Infof("Updated S3 bucket policy - %s", globalAWSBucketName)
+				}
+			})
+			// Create BackupLocation with bucketWithPolicy and SSE set to false
+			Step("Create BackupLocation with bucketWithOutPolicy and SSE set to false", func() {
+				log.InfoD(fmt.Sprintf("Create BackupLocation with bucketWithOutPolicy and SSE set to false"))
+				cloudCredUidForBlWithoutSse = uuid.New()
+				cloudCredUidList = append(cloudCredUidList, cloudCredUID)
+				backupLocationUidWithoutSse = uuid.New()
+				credNameForBlWithoutSse = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
```

Comment:
where are validating the backup failure with this backuplocation ?

Reply:
Done

Reply:
Done ,Added negative scenario